crates/nu-command/src/classified/external.rs

```diff
@@ -172,9 +164,29 @@ fn spawn_cmd_command(command: &ExternalCommand, args: &[String]) -> Command {
     process
 }
 
+fn has_unsafe_shell_characters(arg: &str) -> bool {
+    lazy_static! {
```

Comment:
please excuse my ignorance here, but why is lazy_static required here. I thought lazy_static ws for consts or globals or something like that?

Reply:
The reason we want it is:
1. We don't want to compile the regex every time it has to be used. It will be used a bunch of times (once for every external command argument), and it's always the same regex.
2. It cannot be a regular static. I guess this is because it allocates internally? It must, given that regexes can be arbitraily complex but I can't say for sure.

This pattern is actually documented in the regex crate https://github.com/rust-lang/regex#usage-avoid-compiling-the-same-regex-in-a-loop ! And I must say, that's where I got the idea of using `lazy_static` from, as I'm a rust beginner.